### PR TITLE
feat: immutable admin structure

### LINKED ISSUE #25: redo admin structure
Desired structure:

## Withdrawal Admin
Permissions
- Can withdraw ETH and ERC20s from Zenith contract. 
- Can transfer its role.

Setup
- Configured once at contract deploy time. 
- Transferrable by self afterwards.

## Sequencer Admin
Permissions
- Can set/remove Sequencer keys.
- Can tansfer its role.

Setup
- Configured once at contract deploy time. 
- Transferrable by self afterwards.

## Sequencer Keys
Permissions 
- Signature is accepted as a stamp that co-signs block submission by Builders.
- CANNOT set/remove/transfer its role. (See: Sequencer Admin above)

Setup
- Configured any time by Sequencer Admin.

### PR DESCRIPTION
**Pull request by anna-carroll**
Closes #25

### CODE REVIEW

File: src/Zenith.sol [hunk 1]
```diff
@@ -67,12 +73,36 @@ contract Zenith is Passage {
...
     event BlockData(bytes blockData);
 
-    /// @notice Initializes the Admin role.
-    /// @dev See `AccessControlDefaultAdminRules` for information on contract administration.
-    ///      - Admin role can grant and revoke Sequencer roles.
-    ///      - Admin role can be transferred via two-step process with a 1 day timelock.
-    /// @param admin - the address that will be the initial admin.
-    constructor(uint256 defaultRollupChainId, address admin) Passage(defaultRollupChainId, admin) {}
+    /// @notice Emitted when a sequencer is added or removed.
+    event SequencerSet(address sequencer, bool permissioned);
```
Comment (prestwich) [inline src/Zenith.sol:77]: indexed?

File: src/Zenith.sol [hunk 2]
```diff
@@ -67,12 +73,36 @@ contract Zenith is Passage {
...
+        emit SequencerSet(sequencer, true);
+    }
+
+    /// @notice Remove a sequencer from the permissioned sequencer list.
+    /// @param sequencer - the address of the sequencer to remove.
+    /// @custom:emits SequencerSet if the sequencer is removed.
+    /// @custom:reverts OnlySequencerAdmin if the caller is not the sequencerAdmin.
+    function removeSequencer(address sequencer) external {
+        if (msg.sender != sequencerAdmin) revert OnlySequencerAdmin();
+        if (!isSequencer[sequencer]) return;
```
Comment (prestwich) [inline src/Zenith.sol:102]: probably unnecessary and may add an extra SLOAD

File: src/Zenith.sol [hunk 3]
```diff
@@ -67,12 +73,36 @@ contract Zenith is Passage {
...
+        sequencerAdmin = _sequencerAdmin;
+    }
+
+    /// @notice Add a sequencer to the permissioned sequencer list.
+    /// @param sequencer - the address of the sequencer to add.
+    /// @custom:emits SequencerSet if the sequencer is added.
+    /// @custom:reverts OnlySequencerAdmin if the caller is not the sequencerAdmin.
+    function addSequencer(address sequencer) external {
+        if (msg.sender != sequencerAdmin) revert OnlySequencerAdmin();
+        if (isSequencer[sequencer]) return;
```
Comment (prestwich) [inline src/Zenith.sol:91]: probably unnecessary and may add an extra SLOAD

**Review by prestwich — CHANGES_REQUESTED** [review]

**Comment by prestwich** [comment]
nits only. also open to approving it this way

File: src/Zenith.sol (hunk 2 above)
Comment (anna-carroll) [reply src/Zenith.sol:102]: it keeps the event history clean; only emitted when there's an actual change

File: src/Zenith.sol (hunk 3 above)
Comment (anna-carroll) [reply src/Zenith.sol:91]: ditto above

File: src/Zenith.sol (hunk 2 above)
Comment (anna-carroll) [reply src/Zenith.sol:102]: happy to spend the extra gas to keep the event history clean since it's an admin function that will only be used occasionally

**Review by prestwich — APPROVED** [review]

File: src/Zenith.sol (hunk 1 above)
Comment (anna-carroll) [reply src/Zenith.sol]: agh. yes: https://github.com/init4tech/zenith/pull/29/commits/a7db3ef75ac2f16cba79dfb5785d58c2e8279dbf

**Pull request merged**